apps/rundown/src/cli/BaseReportAction.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import {
+  CommandLineAction,
+  ICommandLineActionOptions,
+  CommandLineStringParameter,
+  CommandLineStringListParameter,
+  CommandLineFlagParameter
+} from '@rushstack/ts-command-line';
+
+export abstract class BaseReportAction extends CommandLineAction {
+  protected scriptParameter: CommandLineStringParameter;
+  protected argsParameter: CommandLineStringListParameter;
+  protected quietParameter: CommandLineFlagParameter;
+  protected ignoreExitCodeParameter: CommandLineFlagParameter;
+
+  public constructor(options: ICommandLineActionOptions) {
+    super(options);
+  }
+
+  // abstract
+  protected onDefineParameters(): void {
+    this.scriptParameter = this.defineStringParameter({
+      parameterLongName: '--script',
+      parameterShortName: '-s',
+      argumentName: 'PATH',
+      description: 'The path to a .js file that will be the entry point for the target Node.js process',
+      required: true
+    });
+    this.argsParameter = this.defineStringListParameter({
+      parameterLongName: '--arg',
+      parameterShortName: '-a',
+      argumentName: 'STRING',
+      description: 'Specifies command-line arguments to be passed to the target Node.js process'
+    });
```

Comment:
This is going to be pretty cumbersome. Doesn't `ts-command-line` support a "rest of the command-line" thing?

Reply:
It does, but it turned out to have a mistake in its design.  I haven't had time to address that yet.

What we could do is pass the entire command-line as a single string. In PR https://github.com/microsoft/rushstack/pull/2060  @sachinjoseph  introduced [string-argv](https://www.npmjs.com/package/string-argv) which seems like the magic parser we need for that.